src/MetaDataDisplay.ts

```diff
@@ -0,0 +1,92 @@
+import {CustomChartOptions} from "./Chart";
+import {countryCodeToName, diseaseCodeToName, vaccineCodeToName} from "./Dictionaries"
+
+// there is almost certainly a library that does this!
+function prettyDataAndTime(): string {
+	const date: Date = new Date();
+	return date.toISOString().substring(0, 19);
+}
+
+// there's a lot of C+P programming here that could arguably be refactored into a
+//single function, but in future they might need to be dictinct functions?
+function prettyCountries(countryArr: string[], maxShow: number): string {
+	const countryCount: number = countryArr.length;
+	if (countryCount > maxShow) {
+		const retStr = countryCount.toString() + " countries including " +
+		               countryArr.slice(0, maxShow).map(countryCodeToName).join(", ");
+		return retStr;
+	} else {
+		const retStr = "Countries: " + countryArr.map(countryCodeToName).join(", ");
+		return retStr;
+	}
+}
+
+function prettyVaccines(vaccineArr: string[], maxShow: number): string {
+	const vaccineCount: number = vaccineArr.length;
+	if (vaccineCount > maxShow) {
+		const retStr = vaccineCount.toString() + " vaccines including " +
+		               vaccineArr.slice(0, maxShow).map(vaccineCodeToName).join(", ");
+		return retStr;
+	} else {
+		const retStr = "Vaccines: " + vaccineArr.map(vaccineCodeToName).join(", ");
+		return retStr;
+	}	
+}
+
+function prettyTouchstones(touchstoneArr: string[], maxShow: number): string {
+	const touchstoneCount: number = touchstoneArr.length;
+	if (touchstoneCount > maxShow) {
+		const retStr = touchstoneCount.toString() + " touchstones including " +
+		               touchstoneArr.slice(0, maxShow).join(", ");
+		return retStr;
+	} else {
+		const retStr = "Touchstones: " + touchstoneArr.join(", ");
+		return retStr;
+	}	
+}
+
+function prettyYears(yearLo: number, yearHi: number): string {
+	const retStr = "From " + yearLo.toString() + " to " + yearHi.toString();
+	return retStr;
+}
+
+function prettyActivities(activityTypes: string[]) {
+	return activityTypes.map(x => x.charAt(0).toUpperCase() + x.slice(1)).join(", ");
+}
+
+export function MetaDataDisplay(chartOptions: CustomChartOptions,
+	                            human: boolean): string {
+	const maxPlot: number = chartOptions.maxPlot;
+
+	const pltType: string = "Impact / Time series";
+	const cumul: string = "cum:TRUE / cum:FALSE"
+
+	if (human) { // produce human readable metadata
+		if (chartOptions.plotType === "Impact") {
+			const metaStr : string = "This plot shows the " +
```

Comment:
It looks like the first sentence for both impact and time series ("This plot shows the..") should be the same, so could just have the code to calculate that once and then append the second sentence depending on plot type. The string construction could be more readable using template strings.